commit 9aa91c5a30f2fee5f86689fbb21e842f53578875
Author: Gottfried Haider <gottfried.haider@gmail.com>
Date:   Tue Sep 26 11:59:51 2017 +0000

    pwm: Set class for exported channels in sysfs
    
    Notifications for devices without bus or class set get dropped by
    dev_uevent_filter(). Adding the class to the exported child matches
    what the GPIO subsystem is doing.
    
    With this change exporting a channel triggers a udev event, which
    gives userspace a chance to fixup permissions and makes it possible
    for non-root users to make use of the PWM subsystem.
    
    Signed-off-by: Gottfried Haider <gottfried.haider@gmail.com>
    CC: Thierry Reding <thierry.reding@gmail.com>
    CC: H Hartley Sweeten <hsweeten@visionengravers.com>
    CC: linux-pwm@vger.kernel.org
    CC: linux-arm-kernel@lists.infradead.org
    CC: linux-rpi-kernel@lists.infradead.org
    Signed-off-by: Thierry Reding <thierry.reding@gmail.com>

diff --git a/drivers/pwm/sysfs.c b/drivers/pwm/sysfs.c
index a813239300c3..83f2b0b15712 100644
--- a/drivers/pwm/sysfs.c
+++ b/drivers/pwm/sysfs.c
@@ -263,6 +263,7 @@ static int pwm_export_child(struct device *parent, struct pwm_device *pwm)
 	export->pwm = pwm;
 	mutex_init(&export->lock);
 
+	export->child.class = parent->class;
 	export->child.release = pwm_export_release;
 	export->child.parent = parent;
 	export->child.devt = MKDEV(0, 0);
